### PR TITLE
Huebot django app integrate with Hue Server

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Bot server integration with existing Hue server currently serving /slack/events.

### CODE REVIEW

File: desktop/core/src/desktop/urls.py [hunk 1]
```diff
@@ -274,3 +274,7 @@
   urlpatterns += [
     url(r'^oidc/', include('mozilla_django_oidc.urls')),
   ]
+
+urlpatterns += [
```
Comment (romainr) [inline desktop/core/src/desktop/urls.py:278]: Let's add a config flag to globally turn on/off the Slack integration.

We could do a new [[slack]] section in [desktop], similarly to https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/conf.py#L685

**Review by romainr — COMMENTED** [review]
Nice!

File: desktop/core/src/desktop/lib/botserver/admin.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/admin.py:1]: Adding it in all files

File: desktop/core/src/desktop/lib/botserver/apps.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/apps.py:18]: Commented it.

File: desktop/core/src/desktop/urls.py (hunk 1 above)
Comment (Harshg999) [reply desktop/core/src/desktop/urls.py]: Added the config flag. Please let me know if my variable naming or messages (like help etc.) are good enough.

Comment (Harshg999) [reply desktop/core/src/desktop/urls.py]: Added the config flag

File: desktop/conf/pseudo-distributed.ini.tmpl [hunk 4]
```diff
@@ -816,6 +816,14 @@
    ## collection_interval=30000
```
Comment (romainr) [inline desktop/conf/pseudo-distributed.ini.tmpl:817]: For info there is also a similar config file to update https://github.com/cloudera/hue/blob/master/desktop/conf.dist/hue.ini
(the production one)

File: desktop/core/src/desktop/lib/botserver/views.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from django.shortcuts import render
+from django.http import HttpResponse
+
+# Create your views here.
+
+def home(request):
+  return HttpResponse("Hello World")
```
[the remark this answers is not in the document]
Comment (romainr) [reply desktop/core/src/desktop/lib/botserver/views.py:25]: Good, let's go with this version and iterate on it!

nit: 'home' is probably not the best name

[the remark this answers is not in the document]
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/views.py:25]: Working on the bot code and will update it soon

File: desktop/core/src/desktop/lib/botserver/views.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+from django.shortcuts import render
+from django.http import HttpResponse
+
+# Create your views here.
+
+def home(request):
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/views.py:24]: Done

File: desktop/conf/pseudo-distributed.ini.tmpl (hunk 4 above)
Comment (Harshg999) [reply desktop/conf/pseudo-distributed.ini.tmpl:817]: Oh, will update it too. Thanks @romainr !

**Pull request merged**